tooling/lib/flatten-schema.js

```diff
@@ -11,14 +11,28 @@ const getName = (parent, prop) => {
 
 const capitalizeFirstLetter = (string = '') => string.charAt(0).toUpperCase() + string.slice(1);
 
-module.exports = (schema, oas) => {
+module.exports = async (schema, oas) => {
+  const derefSchema = await $RefParser.dereference(
+    { ...schema, ...oas },
+    {
+      dereference: {
+        // If circular `$refs` are ignored they'll remain in `derefSchema` as `$ref: String`, otherwise `$ref‘ just
+        // won't exist. This allows us to do easy circular reference detection.
+        circular: 'ignore',
```

Comment:
What would be the way we would detect whether a dereferenced object after this function call? Whether a `$ref: String` exists within the dereferenced object?

Reply:
NVM I see it now: https://github.com/readmeio/oas/pull/292/files#diff-204487ea7066d929383376809d6e70ea36310eca01a89b9ef90cc8e7d366f150R35

Reply:
If `$ref` exists in `derefSchema` at all, then it's circular.

Reply:
Gotcha. Are there any other times where a $ref cannot be satisfied?

Reply:
If it's an external ref (like a URL to another spec) it'll remain unsatisfied if it's inaccessible.

That said, we shouldn't be attempting to resolve external refs at this point in the application cycle so I'm going to add `resolve.external: false` into the options here to prevent that as well as a test for that.